src/components/custom/ZwaveGraph.vue

```diff
@@ -0,0 +1,990 @@
+          this.hammer = Hammer(options.svgElement, {
+            inputClass: Hammer.SUPPORT_POINTER_EVENTS
+              ? Hammer.PointerEventInput
+              : Hammer.TouchInput
+          })
+
+          // Enable pinch
+          this.hammer.get('pinch').set({
+            enable: true
+          })
+
+          // Handle double tap
+          this.hammer.on('doubletap', function () {
+            instance.zoomIn()
+          })
+
+          // Handle pan
+          this.hammer.on('panstart panmove', function (ev) {
+            // On pan start reset panned variables
+            if (ev.type === 'panstart') {
+              pannedX = 0
+              pannedY = 0
+            }
+
+            // Pan only the difference
+            instance.panBy({
+              x: ev.deltaX - pannedX,
+              y: ev.deltaY - pannedY
+            })
+            pannedX = ev.deltaX
+            pannedY = ev.deltaY
+          })
+
+          // Handle pinch
+          this.hammer.on('pinchstart pinchmove', function (ev) {
+            // On pinch start remember initial zoom
+            if (ev.type === 'pinchstart') {
+              initialScale = instance.getZoom()
+              instance.zoomAtPoint(initialScale * ev.scale, {
+                x: ev.center.x,
+                y: ev.center.y
+              })
+            }
+
+            instance.zoomAtPoint(initialScale * ev.scale, {
+              x: ev.center.x,
+              y: ev.center.y
+            })
+          })
+
+          // Prevent moving the page on some devices when panning over SVG
+          options.svgElement.addEventListener('touchmove', function (e) {
+            e.preventDefault()
+          })
+        }
+      }
+    }
+  },
+  watch: {
+    nodes () {
+      this.paintGraph()
+    }
+  },
+  mounted () {},
+  beforeDestroy () {
+    this.hammer.destroy()
+  },
+  methods: {
+    paintGraph () {
+      this.$refs.svg.innerHTML = ''
+      this.$refs.miniSvg.innerHTML = ''
+
+      // https://github.com/dagrejs/dagre/wiki#using-dagre
+      const g = new dagreD3.graphlib.Graph({
+        compound: true
+      }).setGraph({})
+      g.graph().rankDir = 'BT'
+      g.graph().nodesep = 10
+      g.graph().ranker = this.ranker
+
+      // Create the renderer
+      // eslint-disable-next-line new-cap
+      const render = new dagreD3.render()
+
+      const svg = d3.select(this.$refs.svg)
+      const inner = svg
+        .append('g')
+        .attr('transform', 'translate(20,200)scale(1)')
+
+      g.graph().minlen = 0
+
+      // Add our custom shape (a house)
+      render.shapes().house = this.renderHouse
+      render.shapes().battery = this.renderBattery
+
+      const { edges, nodes } = this.listNodes()
+      // Set the parents to define which nodes belong to which cluster
+
+      // add nodes  to graph
+      for (let i = 0; i < nodes.length; i++) {
+        const node = nodes[i]
+        g.setNode(node.id, node)
```

Comment:
@robertsLando according to <https://github.com/dagrejs/graphlib/wiki/API-Reference#setNode> the second parameter to `setNode()` is the label, so I think it should be `g.setNode(node.id, node.id)` instead.

Reply:
nope, label can be an object so it's correct to use the node object there AFAIK. Did you try to use a string instead?

Reply:
You're right. Yes, I've tried and the graph vanished ;-).

Reply:
Was just a misleading documentation to name it "label" if it can be an object.

Reply:
So maybe <https://github.com/dagrejs/graphlib/wiki/API-Reference#setDefaultNodeLabel> can be used to set the default label for nodes. I'll try.